### PR TITLE
BRIDGE-636 Redesign tapping activity to support bilateral tapping task

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
This is a fairly significant change to AppCore. I added unit testing around adding in an activity to test tapping with each hand as well as archiving that data. I removed APCDataArchiver (it is not referenced anywhere) and added APCTaskResultArchiver. This base class is the default archiver for APCBaseTaskViewController.  

Please let me know if you need a tour. It's a lot of changes.

### CODE REVIEW

**Comment by EricSiegNW** [comment]
The bilateral tapping activity code all looked fine. I haven't wrapped my head around the implications/necessity of the APCTaskResultArchiver yet and how it compares to APCDataArchiverAndUploader yet.

File: APCAppCore/APCAppCore/Library/Categories/ORKFileResult+Filename.m [hunk 1]
```diff
@@ -33,41 +33,24 @@
...
-        stepIdentifier = @"";
+    if (stepIdentifier != nil) {
+        if (result.length > 0) {
+            [result appendString:@"_"];
+        }
+        [result appendString:stepIdentifier];
     }
     
-    return [fileResultIdentifier stringByAppendingString:stepIdentifier];
+    return [result copy];
```
Comment (Erin-Mounts) [inline APCAppCore/APCAppCore/Library/Categories/ORKFileResult+Filename.m:53]: why not just return result?

Comment (syoung-smallwisdom) [inline APCAppCore/APCAppCore/Library/Categories/ORKFileResult+Filename.m:53]: result is mutable and the method signature is immutable.

File: APCAppCore/APCAppCore/Library/Categories/ORKOrderedTask+APCHelper.m [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    NSMutableArray *steps = [orkTask.steps mutableCopy];
+    
+    // Look for the tapping instrution step
+    const NSUInteger tappingInstructionIdx = 1;
+    ORKInstructionStep *tappingInstructionStep = nil;
+    if ((orkTask.steps.count > tappingInstructionIdx) &&
+        [orkTask.steps[tappingInstructionIdx] isKindOfClass:[ORKInstructionStep class]])
+    {
+        tappingInstructionStep = (ORKInstructionStep *)orkTask.steps[tappingInstructionIdx];
+        tappingInstructionStep.detailText = NSLocalizedStringWithDefaultValue(@"Step Detail Text", @"APCAppCore", APCBundle(),
```
Comment (Erin-Mounts) [inline APCAppCore/APCAppCore/Library/Categories/ORKOrderedTask+APCHelper.m]: I'd use a more specific key to avoid potential collisions in the future.

**Comment by syoung-smallwisdom** [comment]
@EricSiegNW 
WRT 'APCDataArchiverAndUploader' verse 'APCDataArchiverAndUploader', they were doing two different things. It is _very_ confusing. 'APCDataArchiver' was unused so I deleted it. It duplicated code in 'APCBaseTaskViewController' called by '-(void)archiveResults' to take a 'ORKTaskResult' and put the information from that into a freshly instantiated 'APCDataArchive' (not to be confused with 'APCDataArchiver'). Then the base class implementation for archiving a tapping result or a memory game result, were empty methods. There was an assumption that subclasses of 'APCBaseViewController' used to implement a memory game or tapping activity would archive the results themselves. This didn't really make sense so I moved the logic for doing that into the base implementation. I moved to using the 'APCTaskResultArchiver' for simpler unit test injection and to simplify any subclassing hierarchy implemented by the apps that use this.

File: APCAppCore/APCAppCore/Library/Categories/ORKFileResult+Filename.m (hunk 1 above)
Comment (Erin-Mounts) [inline APCAppCore/APCAppCore/Library/Categories/ORKFileResult+Filename.m:53]: oh yeah derp.

File: APCAppCore/APCAppCore/Library/DataArchiverAndUploader/APCDataArchive.h [hunk 3]
```diff
@@ -89,7 +89,18 @@
  
  @param     filename                Filename for the json data to be included without path extension
  */
-- (void)insertIntoArchive:(NSDictionary *)dictionary filename: (NSString *)filename;
+- (void)insertDictionaryIntoArchive:(NSDictionary *)dictionary filename: (NSString *)filename;
```
Comment (Erin-Mounts) [inline APCAppCore/APCAppCore/Library/DataArchiverAndUploader/APCDataArchive.h:92]: better name is better

**Comment by Erin-Mounts** [comment]
one nitpicky comment on a localization key, one dumb question you already answered, otherwise looks great.

File: APCAppCore/APCAppCore/Library/DataArchiverAndUploader/APCDataArchive.h (hunk 3 above)
Comment (syoung-smallwisdom) [inline APCAppCore/APCAppCore/Library/DataArchiverAndUploader/APCDataArchive.h:92]: Yeah, I got very confused about why it wasn't crashing on attempting to push a dictionary when expecting NSData.

**Pull request merged**